plugins/modules/docker_compose_v2.py

```diff
@@ -0,0 +1,926 @@
+  type: str
+stderr:
+  description:
+    - The stderr from docker-compose.
+  returned: always, unless when C(docker-compose) was not given the chance to run
+  type: str
+containers:
+  description:
+    - A dictionary mapping the various status of containers during C(docker-compose) operation.
+  returned: always, unless when C(docker-compose) was not given the chance to run
+  type: complex
+  contains:
+    container_name:
+      description: Name of the container.
+      returned: always, unless when C(docker-compose) was not given the chance to run
+      type: list
+      elements: str
+      example: ["stopped", "removed"]
+volumes:
+  description:
+    - A dictionary mapping the various status of volumes during C(docker-compose) operation.
+  returned: always, unless when C(docker-compose) was not given the chance to run
+  type: complex
+  contains:
+    volume_name:
+      description: Name of the volume.
+      returned: always, unless when C(docker-compose) was not given the chance to run
+      type: list
+      elements: str
+      example: ["created"]
+images:
+  description:
+    - A dictionary mapping the various status of volumes during C(docker-compose) operation.
+  returned: always, unless when C(docker-compose) was not given the chance to run
+  type: complex
+  contains:
+    image_name:
+      description: Name of the image.
+      returned: always, unless when C(docker-compose) was not given the chance to run
+      type: list
+      elements: str
+      example: ["removed"]
+networks:
+  description:
+    - A dictionary mapping the various status of networks during C(docker-compose) operation.
+  returned: always, unless when C(docker-compose) was not given the chance to run
+  type: complex
+  contains:
+    image_name:
+      description: Name of the image.
+      returned: always, unless when C(docker-compose) was not given the chance to run
+      type: list
+      elements: str
+      example: ["created"]
+'''
+
+
+import sys
+if sys.version_info[0] == 3 and sys.version_info[1] >= 5:
+    from typing import List, Optional, Tuple, Union, FrozenSet, Dict, Type, TYPE_CHECKING, Any
+    if sys.version_info[1] >= 8:
+        from typing import Literal, Final
+    else:
+        try:
+            from typing_extensions import Literal, Final
+        except ImportError:
+            pass
+    # `Text` can be removed and replaced by `str` once Python 2 support is dropped
+    if sys.version_info[1] < 11:
+        # pylint: disable-next=deprecated-class
+        from typing import Text
+    else:
+        Text = str
+else:
+    TYPE_CHECKING = False
+import re
+from collections import defaultdict, namedtuple
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible_collections.community.docker.plugins.module_utils.util import (
+    DOCKER_COMMON_ARGS,
+)
+from ansible.module_utils.common.yaml import HAS_YAML, yaml_dump
```

Comment:
Please note that this helper was only added in ansible-core 2.12. My suggestion would be to vendor https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/common/yaml.py as `plugins/module_utils/_yaml.py` in this collection, with a comment that it can be replaced by `ansible.module_utils.common.yaml` once support for ansible-core 2.11 has been dropped.

Reply:
Vendored